music21/base.py

```diff
@@ -1278,6 +1289,29 @@ def getContextByClass(
         >>> b.next('Note')
         <music21.note.Note C>
 
+        A Stream might contain several elements at the same offset, leading to
+        potentially surprising results where searching "atOrBefore" does not search
+        an element that is technically the NEXT node in the tree but still at 0.0:
+
+        >>> s = stream.Stream()
+        >>> s.insert(0, clef.BassClef())
+        >>> s.next()
+        <music21.clef.BassClef>
+        >>> s.getContextByClass(clef.Clef) is None
+        True
+        >>> s.getContextByClass(clef.Clef, getElementMethod='getElementAtOrAfter')
+        <music21.clef.BassClef>
+
+        This can be remedied by explicitly searching by offsets:
+
+        >>> s.getContextByClass(clef.Clef, getElementMethod='getElementAtOrBeforeOffset')
+        <music21.clef.BassClef>
+
+        Or by not limiting the search by temporal position at all:
+
+        >>> s.getContextByClass(clef.Clef, getElementMethod='all')
```

Comment:
These should be updated to the new Enum, along with the doc on line 1293.

Reply:
(we can test the string version in a unittest, but better to get it out of the doc.